util/chplenv/defaults.py

```diff
@@ -0,0 +1,180 @@
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+from utils import chplvars, memoize
+
+
+class ChapelConfig(object):
+    """ Class for parsing chplconfig file and providing 'get' interface """
+    def __init__(self):
+        """ Find and parse chplconfig file, populating self.chplconfig """
+
+        # Dictionary containing all of the env vars defined in chplconfig
+        self.chplconfig = {}
+
+        # List of warnings to track if a warning has occurred
+        self.warnings = []
+
+        # Find chplconfigfile and prettypath
+        self.find()
+
+        # Populate chplconfig
+        if self.chplconfigfile:
+            self.parse()
+
+        # Print any warnings accumulated throughout constructor
+        if self.warnings:
+            self.printwarnings()
+
+    def get(self, var):
+        """ Wrapper for chplconfig[var] intended to mirror os.environ.get() """
+        if var in self.chplconfig.keys():
+            return self.chplconfig[var]
+
+    def find(self):
+        """ Find chplconfig file path"""
+
+        # Places to look for a chplconfig file
+        chplconfigpath = os.environ.get('CHPL_CONFIG')
+        home = os.path.expanduser('~')
+
+        # Construct path to chplconfig file, ccpath
+        if chplconfigpath:
+            ccpath = chplconfigpath
+            self.prettypath = '$CHPL_CONFIG/'
+        else:
+            ccpath = home
+            self.prettypath = '~/'
+
+        # Search for both visible and hidden files
+        visible = os.path.join(ccpath, 'chplconfig')
+        hidden = os.path.join(ccpath, '.chplconfig')
+
+        # Set self.chplconfigpath if it exists
+        if os.path.isfile(visible):
+            self.chplconfigfile = visible
+            self.prettypath += 'chplconfig'
+        elif os.path.isfile(hidden):
+            self.chplconfigfile = hidden
+            self.prettypath += '.chplconfig'
+        else:
+            self.chplconfigfile = None
+            if chplconfigpath:
+                self.warnings.append(
+                (
+                    'Warning: No chplconfig or .chplconfig file is found in '
+                    'the defined $CHPL_CONFIG\n'
+                ))
+
+    def parse(self):
+        """ Parse chplconfig file for acceptable env var overrides """
+
+        # Parse the chplconfig file and populate the chplconfig dict
+        with open(self.chplconfigfile, 'r') as ccfile:
+
+            # Split lines into fields delimited by '='
+            linefields = [l.split('=') for l in ccfile.readlines()]
+
+            for linenum, fields in enumerate([lf for lf in linefields]):
+
+
+                if self.invalid_assignment(fields, linenum):
+                    continue
+
+                var, val = [f.strip() for f in fields]
+
+                if self.invalid_entry(var, val, linenum):
+                    continue
+
+                self.chplconfig[var] = val
+
+    def invalid_assignment(self, fields, linenum):
```

Comment:
`invalid_assignment` is a slightly odd name to me. I'd prefer `is_invalid_assignment` same with `invalid_entry`